internal/confmapprovider/discovery/discoverer.go

```diff
@@ -117,6 +119,38 @@ func newDiscoverer(logger *zap.Logger) (*discoverer, error) {
 	return d, nil
 }
 
+func (d *discoverer) discoveryAuthConfFromEnv(discoveryReceiverRaw map[string]any) (*confmap.Conf, error) {
+	yamlProvider := yamlprovider.New()
+	envProvider := envprovider.New()
+
+	providers := map[string]confmap.Provider{
+		"yaml": yamlProvider,
+		"env":  envProvider,
+	}
+	out, err := yaml.Marshal(discoveryReceiverRaw)
+	if err != nil {
+		d.logger.Warn(fmt.Sprintf("Failed to create resolver from yaml provider. %v", err))
+	}
+	uris := []string{fmt.Sprintf("yaml:%s", out[:])}
+	resolver, err := confmap.NewResolver(confmap.ResolverSettings{
+		URIs:       uris,
+		Providers:  providers,
+		Converters: []confmap.Converter{d.expandConverter},
+	})
+	if err != nil {
+		d.logger.Error(fmt.Sprintf("Failed to create a resolver from the given uris:%s and providers:%v. %v", uris, providers, err))
```

Comment:
A lot of the config structs use https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/configopaque/opaque.go but I think this still can log secrets. Maybe just include the err to restrict what comes out? (not sure if the err will contain the full input though).

Reply:
err doesn't include the secrets. I will just log err and yes providers do not give us much in this case.